app/helpers/courses_helper.rb

```diff
@@ -49,4 +49,14 @@ def percent_complete(course)
     end
     ""
   end
+
+  def percent_complete_without_user(course, lesson_id)
+    session[:completed_lessons] ||= []
+    total_lessons = course.lessons.count
+    session[:completed_lessons] << lesson_id unless session[:completed_lessons].include?(lesson_id)
+    completed = session[:completed_lessons].count
+    return 0 if total_lessons == 0
```

Comment:
I just realized there is already a "percent_complete" method, so maybe it should be called something like "calc_percent_complete" and just live as it's own little purely functional method.